src/discussions/in-context-topics/TopicPostsView.test.jsx

```diff
@@ -0,0 +1,280 @@
+
+      await act(async () => {
+        setupPostsMockResponse(parent.id, 3);
+        renderComponent({ topicId: parent.id });
+      });
+
+      await waitFor(async () => {
+        const backButton = await screen.getByLabelText('Back to topics list');
+        const parentHeader = await screen.findByText(parent.name);
+
+        expect(backButton).toBeInTheDocument();
+        expect(parentHeader).toBeInTheDocument();
+      });
+    });
+
+  it('A back button should redirect from list of posts to list of units.', async () => {
+    await setupTopicsMockResponse();
+    const subSection = coursewareTopics[0].children[0];
+    const unit = subSection.children[0];
+
+    await act(async () => {
+      setupPostsMockResponse(unit.id, 2);
+      renderComponent({ topicId: unit.id });
+    });
+
+    const backButton = await screen.getByLabelText('Back to topics list');
+
+    await act(async () => fireEvent.click(backButton));
+    await waitFor(async () => {
+      renderComponent({ topicId: undefined, category: subSection.id });
+
+      const subSectionList = await container.querySelector('.list-group');
+      const units = await subSectionList.querySelectorAll('.discussion-topic');
+      const unitHeader = await within(subSectionList).queryByText(unit.name);
+
+      expect(lastLocation.pathname.endsWith(`/category/${subSection.id}`)).toBeTruthy();
+      expect(unitHeader).toBeInTheDocument();
+      expect(units).toHaveLength(4);
+    });
+  });
+
+  it('A back button should redirect from units to the parent/selected subsection.', async () => {
+    await setupTopicsMockResponse();
+    const subSection = coursewareTopics[0].children[0];
+
+    renderComponent({ topicId: undefined, category: subSection.id });
+
+    const backButton = await screen.getByLabelText('Back to topics list');
+
+    await act(async () => fireEvent.click(backButton));
+    await waitFor(async () => {
+      renderComponent();
+
+      const sectionList = await container.querySelector('.list-group');
+      const subSections = await sectionList.querySelectorAll('.discussion-topic-group');
+      const subSectionHeader = await within(sectionList).queryByText(subSection.displayName);
+
+      expect(lastLocation.pathname.endsWith('/topics/')).toBeTruthy();
+      expect(subSectionHeader).toBeInTheDocument();
+      expect(subSections).toHaveLength(3);
+    });
+  });
+
+  test.each([
+    { searchText: 'hello world', output: 'Showing 0 results for' },
+    { searchText: 'introduction', output: 'Showing 8 results for' },
+  ])('It should have a search bar with a clear button and \'$output\' results found text.',
+    async ({ searchText, output }) => {
+      await setupTopicsMockResponse();
+      await renderComponent();
+
+      const searchField = await within(container).getByPlaceholderText('Search topics');
+      const search = await container.querySelector('.pointer-cursor-hover');
+      const searchButton = await search.querySelector('.pgn__icon');
```

Comment:
Please do not use paragon classes in test cases. add test_id to directly get the element.